protobuf/kernel.proto

```diff
@@ -2,312 +2,201 @@ syntax = "proto3";
-    Transaction Transaction = 2;
-    SignatureStatus SignatureStatus = 3;
-    RefBlockStatus RefBlockStatus = 4;
-    TransactionStatus TransactionStatus = 5;
-    bool IsSystemTxn = 6;
-    int64 ExecutedBlockNumber = 7;
+    Hash transaction_id = 1;
+    Transaction transaction = 2;
+    SignatureStatus signature_status = 3;
+    RefBlockStatus ref_block_status = 4;
+    TransactionStatus transaction_status = 5;
+    bool is_system_txn = 6;
+    int64 executed_block_number = 7;
 }
 
-//message StatePath {
-//    repeated string parts = 1;
-//}
-
-//message ScopedStatePath {
-//    Address address = 1;
-//    StatePath path = 2;
-//}
-
 message StateValue {
-    bytes CurrentValue = 1;
-    bytes OriginalValue = 2;
+    bytes current_value = 1;
+    bytes original_value = 2;
 }
 
 message StateChange {
-    StatePath StatePath = 1;
-    StateValue StateValue = 2;
+    StatePath state_path = 1;
+    StateValue state_value = 2;
 }
 
-//enum TransactionResultStatus {
-//    NotExisted = 0;
-//    Pending = 1;
-//    Failed = 2;
-//    Mined = 3;
-//    Unexecutable = 4;
-//}
-//
-//message TransactionResult {
-//    Hash TransactionId = 1;
-//    TransactionResultStatus Status = 2;
-//    repeated LogEvent Logs = 3;
-//    bytes Bloom = 4;
-//    bytes ReturnValue = 5;
-//    int64 BlockNumber = 6;
-//    Hash BlockHash = 7;
-//    int32 Index = 8;
-//    Hash StateHash = 9;
-//    // Merkle proof path for this transaction
-//    // TODO: Remove DeferredTxnId
-//    repeated Transaction DeferredTransactions = 10;
-//    Hash DeferredTxnId = 11;
-//    string Error = 12;
-//    string ReadableReturnValue = 13;
-//}
-
 // For failed transactions, its status follows the Min of
 // its own status and its inline transactions'
 enum ExecutionStatus {
-    Undefined = 0;
+    UNDEFINED = 0;
     // Successful =>
-    Executed = 1;
+    EXECUTED = 1;
 
     // Failed =>
     //   Infrastructure reasons
-    Canceled = -1;
-    SystemError = -2;
+    CANCELED = -1;
+    SYSTEM_ERROR = -2;
 
     //   Contract reasons
-    ContractError = -10;
-    ExceededMaxCallDepth = -11;
-    InsufficientTransactionFees = -12;
+    CONTRACT_ERROR = -10;
+    EXCEEDED_MAX_CALL_DEPTH = -11;
+    INSUFFICIENT_TRANSACTION_FEES = -12;
 
     // Pre-failed
-    Prefailed = -99;
+    PREFAILED = -99;
 }
 
 message TransactionTrace {
-    Hash TransactionId = 1;
-    bytes ReturnValue = 2;
-    string StdOut = 3;
-    string StdErr = 4;
-    repeated Transaction PreTransactions = 5;
-    repeated TransactionTrace PreTraces = 6;
-    repeated Transaction InlineTransactions = 7;
-    repeated TransactionTrace InlineTraces = 8;
-    repeated LogEvent Logs = 9;
-    int64 Elapsed = 10;
-    ExecutionStatus ExecutionStatus = 11;
-    TransactionExecutingStateSet StateSet = 12;
-    bytes DeferredTransaction = 13;
-    string ReadableReturnValue = 14;
+    Hash transaction_id = 1;
+    bytes return_value = 2;
+    string std_out = 3;
+    string std_err = 4;
```

Comment:
rename

Reply:
this has been rename to "Error", cf. #1805